Minio/DataModel/Select/SelectResponseStream.cs

```diff
@@ -0,0 +1,256 @@
+            }
+            this._isProcessing = true;
+            this.payloadStream.Seek(0, SeekOrigin.Begin);
+            this.start();
+        }
+        
+        private void start()
+        {
+            int numBytesRead = 0;
+            int msgOffset = 0;
+            while (_isProcessing)
+            {   
+                var n =  ReadFromStream(prelude);
+                numBytesRead += n;
+                n =  ReadFromStream(preludeCRC);
+                var preludeCRCBytes = preludeCRC.ToArray();
+                if (BitConverter.IsLittleEndian)
+                {
+                    Array.Reverse(preludeCRCBytes); 
+                }
+                numBytesRead += n;
+                var inputArray = new byte[prelude.Length + 4 ];
+                System.Buffer.BlockCopy(prelude,0,inputArray,0, prelude.Length);
+
+                // write real data to inputArray
+                Crc32Algorithm.ComputeAndWriteToEnd(inputArray); // last 4 bytes contains CRC
+                // transferring data or writing reading, and checking as final operation
+                if (!Crc32Algorithm.IsValidWithCrcAtEnd(inputArray))
+                {
+                    throw new ArgumentException("invalid prelude CRC");
+                }
+
+                if (!Enumerable.SequenceEqual(inputArray.Skip(prelude.Length).Take(4), preludeCRCBytes))
+                {
+                    throw new ArgumentException("Prelude CRC Mismatch");
+                }
+                 var bytes = prelude.Take(4).ToArray();
+                 if (BitConverter.IsLittleEndian)
+                 {
+                    Array.Reverse(bytes); 
+                 }
+                int totalLength = BitConverter.ToInt32(bytes, 0);
+                bytes = prelude.Skip(4).Take(4).ToArray();
+                if (BitConverter.IsLittleEndian)
+                {
+                    Array.Reverse(bytes); 
+                }
+
+                int headerLength = BitConverter.ToInt32(bytes, 0);
+                int payloadLength = totalLength - headerLength - 16;
+
+                var headers = new byte[headerLength];
+                var payload  = new byte[payloadLength];
+                int num = ReadFromStream(headers);
+                if (num != headerLength)
+                {
+                    throw new IOException("insufficient data");
+
+                }
+                num = ReadFromStream(payload);
+                if (num != payloadLength)
+                {
+                    throw new IOException("insufficient data");
+                }
+
+                numBytesRead += num;
+                num = ReadFromStream(messageCRC);
+                var messageCRCBytes = messageCRC.ToArray();
+                if (BitConverter.IsLittleEndian)
+                {
+                    Array.Reverse(messageCRCBytes); 
+                }
+                // now verify message CRC
+                inputArray = new byte[totalLength];
```

Comment:
Should we allocate 4 more bytes for the messageCRC check? like we did for the preludeCRC above?